src/main/java/edu/cornell/kfs/coa/businessobject/GlobalObjectWithIndirectCostRecoveryAccounts.java

```diff
@@ -0,0 +1,109 @@
+package edu.cornell.kfs.coa.businessobject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.kuali.kfs.coa.businessobject.IndirectCostRecoveryAccount;
+import org.kuali.rice.krad.bo.GlobalBusinessObjectDetailBase;
+import org.kuali.rice.krad.bo.PersistableBusinessObjectBase;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+public abstract class GlobalObjectWithIndirectCostRecoveryAccounts extends PersistableBusinessObjectBase {
+	protected List<IndirectCostRecoveryAccountChange> indirectCostRecoveryAccounts;
+	
+	public GlobalObjectWithIndirectCostRecoveryAccounts() {
+		indirectCostRecoveryAccounts = new ArrayList<IndirectCostRecoveryAccountChange>();
+	}
+
+	public List<IndirectCostRecoveryAccountChange> getIndirectCostRecoveryAccounts() {
+		return indirectCostRecoveryAccounts;
+	}
+
+	public void setIndirectCostRecoveryAccounts(
+			List<IndirectCostRecoveryAccountChange> indirectCostRecoveryAccounts) {
+		this.indirectCostRecoveryAccounts = indirectCostRecoveryAccounts;
+	}
+	
+    public List<IndirectCostRecoveryAccountChange> getActiveIndirectCostRecoveryAccounts() {
+        List<IndirectCostRecoveryAccountChange> activeList = new ArrayList<IndirectCostRecoveryAccountChange>();
+        for (IndirectCostRecoveryAccountChange icr : getIndirectCostRecoveryAccounts()){
+            if (icr.isActive()){
+                activeList.add(IndirectCostRecoveryAccountChange.copyICRAccount(icr));
+            }
+        }
+        return activeList;
+    }
+	
+	public abstract List<? extends GlobalBusinessObjectDetailBase> getGlobalObjectDetails();
+	
+	public abstract Map<GlobalBusinessObjectDetailBase, List<IndirectCostRecoveryAccount>> getGlobalObjectDetailsAndIcrAccountsMap();
+	
+	public abstract IndirectCostRecoveryAccount createIndirectCostRecoveryAccountFromChange(GlobalBusinessObjectDetailBase globalDetail, IndirectCostRecoveryAccountChange newICR);
+	
+	public boolean hasIcrAccounts(){
+		return ObjectUtils.isNotNull(indirectCostRecoveryAccounts) && indirectCostRecoveryAccounts.size() > 0;
+	}
+	
+	protected void updateIcrAccounts(GlobalBusinessObjectDetailBase globalDetail, List<IndirectCostRecoveryAccount> icrAccounts) {
+		Map<Integer, Integer> alreadyUpdatedIndexes = new HashMap<Integer, Integer>();
+		List<IndirectCostRecoveryAccount> addList = new ArrayList<IndirectCostRecoveryAccount>();
+
+		if (indirectCostRecoveryAccounts.size() > 0) {
+			for (IndirectCostRecoveryAccountChange newICR : indirectCostRecoveryAccounts) {
+				boolean found = false;
+
+				int temp = -1;
+				int i = 0;
+
+				while (i < icrAccounts.size() && (!found || (found && temp!=-1))) {
```

Comment:
This code is in fact the same code you have previously reviewed but refactored into a common class that both AccountGlobal and SubAccountGlobal use.
As I have mentioned to you yesterday on Hipchat, my understanding during our Clean Code meeting was that 'while' is preferred to iteration with 'break'.  The condition for the while loop however I believe can be improved. I have extracted it into a function to improve clarity. The function name should be more descriptive.